### PR TITLE
Workflow listing actions — part 2 [MAILPOET-4540]

### PR DESCRIPTION
**Pull request by JanJakes**
## Description

Based on the original PR by @johnolek, this:
1. Adds a `@wordpress/data`-based store for automation listing.
2. Uses `@wordpress/notices` store to render MailPoet notices (because those have WP listing styles).
3. Uses dropdown menu from `@wordpress/components` for listing actions.
4. Implements duplicate, trash, restore, and delete actions.
5. Implements notices and "undo" buttons.
6. Improves listing rendering and aligns it closer to specs.
7. Improves workflow stats rendering and aligns it closer to specs. Reuses a single component across editor/listing.

The commits also tell the story.

## Code review notes

_N/A_

## QA notes

Automation only, QA step can be skipped.

## Linked PRs

Implements UI for https://github.com/mailpoet/mailpoet/pull/4376.

## Linked tickets

[MAILPOET-4540]

## After-merge notes

_N/A_

[MAILPOET-4540]: https://mailpoet.atlassian.net/browse/MAILPOET-4540?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/assets/js/src/automation/listing/store/selectors.ts [hunk 1]
```diff
@@ -1,6 +1,11 @@
 import { State } from './types';
 import { Workflow } from '../workflow';
+import { workflowCount } from '../../config';
 
 export function getWorkflows(state: State): Workflow[] {
   return state.workflows;
 }
+
+export function getWorkflowCount(state: State): number {
+  return state.workflows ? state.workflows.length : workflowCount;
```
Comment (websupporter) [inline mailpoet/assets/js/src/automation/listing/store/selectors.ts:10]: Curious: Why do we need to "import" the workflowCount. To me it looks like `return state.workflows ? state.workflows.length : 0;` would do the same.

File: mailpoet/assets/js/src/automation/listing/index.tsx [hunk 2]
```diff
@@ -143,21 +147,15 @@ export function AutomationListing({ workflows, loading }: Props): JSX.Element {
               allowFreeTextSearch
```
Comment (websupporter) [inline mailpoet/assets/js/src/automation/listing/index.tsx:150]: Not sure if it's related to this PR, but our search does currently not work. This happens, when I type something in the search box: 
image

File: mailpoet/assets/js/src/automation/listing/store/selectors.ts (hunk 1 above)
Comment (websupporter) [reply mailpoet/assets/js/src/automation/listing/store/selectors.ts:10]: Ah: I think this prevents the `<Onboarding />` to show up while we just load the workflows.

File: mailpoet/assets/js/src/automation/listing/index.tsx (hunk 2 above)
Comment (JanJakes) [reply mailpoet/assets/js/src/automation/listing/index.tsx:150]: Yeah, we never implemented it. I'm thinking maybe we should hide it for the MVP 🤔

File: mailpoet/assets/js/src/automation/listing/store/selectors.ts (hunk 1 above)
Comment (JanJakes) [reply mailpoet/assets/js/src/automation/listing/store/selectors.ts:10]: Yeah, the imported count tells you if you will have any workflows at all and you can decide to show either onboarding or a loader for the listing.

I was thinking to preload even the workflow list in the template but then I started to think about tabs and paging and stuff and just left the count as John did. In his version the count was a separate field in the state but it then creates too many states of "count" vs. "workflows" so I just used it in the selector when workflows are not loaded yet.

File: mailpoet/assets/css/src/components-automation/_statistics.scss [hunk 3]
```diff
@@ -58,6 +59,10 @@
       font-size: 12px;
       font-weight: 400;
       line-height: 16px;
+
+      &.display-after {
+        order: 1;
+      }
```
Comment (websupporter) [inline mailpoet/assets/css/src/components-automation/_statistics.scss]: :clap: 

this is how to solve the problem from the number on one screen is on top, at the other on bottom. Didn't think about a CSS solution when working on it. Thank you!

**Review by websupporter — APPROVED** [review]
Thank you for the PR. Very nice changes. I love especially the fixes for the statistics component :heart:

**Pull request merged**